pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -479,69 +480,60 @@ func isPodTerminating(pod corev1.Pod) bool {
 	return pod.ObjectMeta.DeletionTimestamp != nil
 }
 
-func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) bool {
+func (r *ReconcileJenkinsBaseConfiguration) isRecreatePodNeeded(currentJenkinsMasterPod corev1.Pod, userAndPasswordHash string) []string {
+	var messages []string
 	if userAndPasswordHash != r.jenkins.Status.UserAndPasswordHash {
-		r.logger.Info("User or password have changed, recreating pod")
-		return true
+		messages = append(messages, "User or password have changed, recreating pod")
 	}
 
 	if r.jenkins.Spec.Restore.RecoveryOnce != 0 && r.jenkins.Status.RestoredBackup != 0 {
-		r.logger.Info(fmt.Sprintf("spec.restore.recoveryOnce is set, recreating pod"))
-		return true
+		messages = append(messages, "spec.restore.recoveryOnce is set, recreating pod")
 	}
 
 	if version.Version != r.jenkins.Status.OperatorVersion {
-		r.logger.Info(fmt.Sprintf("Jenkins Operator version has changed, actual '%+v' new '%+v', recreating pod",
+		messages = append(messages, fmt.Sprintf("Jenkins Operator version has changed, actual '%+v' new '%+v', recreating pod",
```

Comment:
split into message and verbose message:

"Jenkins Operator version has changed"

messages = append(messages, fmt.Sprintf("Jenkins Operator version has changed, actual '%+v' new '%+v', recreating pod",